oak_runtime/src/permissions.rs

```diff
@@ -20,29 +20,43 @@
 /// To run an Oak application, one must provide, as input to the `oak_loader`, a TOML
 /// file of permissions, which is parsed into an instance of `PermissionsConfiguration`.
 /// The permissions are applied dynamically at runtime, when creating the Nodes.
-#[allow(dead_code)]
-#[derive(serde::Deserialize, Debug)]
+#[derive(Clone, Default, serde::Deserialize, Debug)]
 #[serde(deny_unknown_fields)]
 pub struct PermissionsConfiguration {
     /// Whether to enable the use of GrpcServerNode.
     #[serde(default)]
-    allow_grpc_server_nodes: bool,
+    pub allow_grpc_server_nodes: bool,
 
     /// Whether to enable the use of HttpServerNode.
     #[serde(default)]
-    allow_http_server_nodes: bool,
+    pub allow_http_server_nodes: bool,
 
     /// Whether to enable the use of LogNode.
     #[serde(default)]
-    allow_log_nodes: bool,
+    pub allow_log_nodes: bool,
 
     /// Whether to allow introspection. If true, the introspection auxiliary service will be
     /// started.
     #[serde(default)]
-    allow_introspection: bool,
+    pub allow_introspection: bool,
 
-    /// List of the allowed URIs that the application can connect to. Only GrpcClientNodes
-    /// that are configured to connect to URIs in this list can be created.
+    /// Whether to enable creating an HTTP client for insecure HTTP connection.
     #[serde(default)]
-    allow_grpc_egress_addresses: Vec<String>,
+    pub allow_insecure_http_egress: bool,
+
+    /// List of the allowed URIs that the application can connect to via a gRPC or an HTTP client.
+    /// Only GrpcClientNodes or HttpClientNodes that are configured to connect to URIs (or
+    /// authorities) in this list can be created.
+    #[serde(default)]
+    pub allow_egress_addresses: Vec<String>,
```

Comment:
Are those full URIs? Or prefixes? Or just hosts? Or authorities? What is the relationship between this and the HTTP egress flag, since these URIs already have a schema AFAICT?

Reply:
Currently, the addresses are `scheme + authority`. But actually, it only works if the scheme is `https`. Perhaps it is better to list the authorities here, instead of the URIs. I have this change in my new commit.  I've also updated the comments with more clarification. 

There is no relation between this list and the `allow_insecure_http_egress` flag. Previously, we decided that we would not distinguish between different insecure HTTP hosts, when creating HttpClient nodes. So it does not make sense to have them in the allow list here.